### ISSUE TITLE
GitHub action publishes the Docker image whenever a PR is open

### ISSUE THREAD

**Issue by tomaka** [substrate-developer-hub/substrate-docs#1051]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Bug report for compiling, code snippets, templates, etc.

[The GitHub Action "build-website-image"](https://github.com/substrate-developer-hub/substrate-docs/blob/d66570f894481b53045121a2e35cb29516c138b8/.github/workflows/check-links.yml#L13-L27) publishes the Docker image whenever a PR is open.
In other words, as soon as someone opens a PR, the changes proposed by the PR are immediately published on Docker Hub, even before the PR is merged.

I suppose that this is not intended, but feel free to close this issue if it is in fact intended.

**Comment by TriplEight** [comment]
The pipeline design indeed seems not optimal, however https://hub.docker.com/u/paritytechno (cc https://github.com/substrate-developer-hub/substrate-docs/issues/415) is not a repository we call "official".

We (CI and Infra teams) control the [following registries](https://github.com/paritytech/ci_cd/wiki/HowTo:-Create-an-image-and-push-it-to-docker-hub-in-a-pipeline#intro). Please take a look at the **Intro** part and the PS there.

Having that said, https://hub.docker.com/u/paritytechno is not something (to my knowledge) is published elsewhere for people to use in the wild (but it's a shame we don't have credentials for it). Thus, it's not a big deal to have it published from a PR, esp knowing that GitHub will fail running this action from an external developer's pipeline as there's no way to allow them using secrets.

Returning to a pipeline design, I would rather suggest building the website and hosting it locally for a further link check than actually publishing it.
